src/test/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/RequesterNameAndAddressTest.java

```diff
@@ -8,94 +8,121 @@
-    }
+        var requestingOrganizationResult = RequesterNameAndAddress.fromString(edifactString);
 
-    @Test
-    void when_buildingSegmentObjectWithoutMandatoryFields_expect_nullPointerException() {
-        assertThrows(NullPointerException.class, () -> RequesterNameAndAddress.builder().build());
+        assertNull(requestingOrganizationResult.getIdentifier());
+        assertNull(requestingOrganizationResult.getRequesterName());
+        assertNull(requestingOrganizationResult.getHealthcareRegistrationIdentificationCode());
+        assertThat(requestingOrganizationResult.getRequestingOrganizationName())
+            .isEqualTo("A GP");
     }
 
     @Test
     void testGetKey() {
-        assertEquals(requesterNameAndAddress.getKey(), "NAD");
+        assertEquals(requester.getKey(), "NAD");
     }
 
     @Test
     void testValidate() {
-        assertDoesNotThrow(requesterNameAndAddress::validate);
+        assertDoesNotThrow(requester::validate);
+        assertDoesNotThrow(requestingOrganization::validate);
 
-        RequesterNameAndAddress emptyIdentifier = new RequesterNameAndAddress(
-            "", HealthcareRegistrationIdentificationCode.GP, "SMITH"
-        );
-        RequesterNameAndAddress emptyRequesterName = new RequesterNameAndAddress(
-            "ABC", HealthcareRegistrationIdentificationCode.GP, ""
-        );
+        RequesterNameAndAddress emptyIdentifier = RequesterNameAndAddress.builder()
+            .identifier(null)
+            .healthcareRegistrationIdentificationCode(HealthcareRegistrationIdentificationCode.GP)
+            .requesterName("SMITH")
+            .requestingOrganizationName(null)
+            .build();
+
+        RequesterNameAndAddress emptyCode = RequesterNameAndAddress.builder()
+                .identifier("ABC")
+                .healthcareRegistrationIdentificationCode(null)
+                .requesterName("SMITH")
+                .requestingOrganizationName(null)
+                .build();
+
+        RequesterNameAndAddress emptyRequesterName = RequesterNameAndAddress.builder()
+                .identifier("ABC")
+                .healthcareRegistrationIdentificationCode(HealthcareRegistrationIdentificationCode.GP)
+                .requesterName(null)
+                .requestingOrganizationName(null)
+                .build();
+
+        RequesterNameAndAddress emptyRequestingOrganizationName = RequesterNameAndAddress.builder()
+                .identifier(null)
+                .healthcareRegistrationIdentificationCode(null)
+                .requesterName(null)
+                .requestingOrganizationName(null)
+                .build();
 
         assertAll(
             () -> assertThatThrownBy(emptyIdentifier::validate)
                 .isExactlyInstanceOf(EdifactValidationException.class)
                 .hasMessage("NAD: Attribute identifier is required"),
+            () -> assertThatThrownBy(emptyCode::validate)
```

Comment:
These feel like separate tests. 

Could we have separate tests for each `validate` scenario i.e. `testValidateMissingIdentifier` and `testValidateEmptyRequestingOrganizationName`?

Reply:
I can do that, I don't mind either way. Strange the `PerformerNameAndAddress` tests are like that and not the `RequesterNameAndAddress` tests